nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -776,6 +797,58 @@ private void executeDML(final ProcessContext context, final ProcessSession sessi
         }
     }
 
+    private void setParameter(PreparedStatement ps, int index, Object value, int fieldSqlType, int sqlType) throws IOException {
+        if (sqlType == Types.BLOB) {
+            // Convert Byte[] into Blob
+            if (fieldSqlType == Types.ARRAY) {
+                if (!(value instanceof byte[])) {
+                    if (value == null) {
+                        try {
+                            ps.setNull(index, Types.BLOB);
+                            return;
+                        } catch (SQLException e) {
+                            throw new IOException("Unable to setNull() on prepared statement" , e);
+                        }
+                    } else {
+                        throw new IOException("Expected BLOB to be of type byte[] but is instead " + value.getClass().getName());
+                    }
+                }
+                byte[] byteArray = (byte[]) value;
+                try (InputStream inputStream = new ByteArrayInputStream(byteArray)) {
+                    ps.setBlob(index, inputStream);
+                } catch (SQLException e) {
+                    throw new IOException("Unable to parse binary data " + value, e.getCause());
+                }
+            } else {
+                try (InputStream inputStream = new ByteArrayInputStream(value.toString().getBytes())) {
+                    ps.setBlob(index, inputStream);
+                } catch (IOException | SQLException e) {
+                    throw new IOException("Unable to parse binary data " + value, e.getCause());
+                }
+            }
+        } else if (sqlType == Types.CLOB) {
+            if (value == null) {
+                try {
+                    ps.setNull(index, Types.CLOB);
+                } catch (SQLException e) {
+                    throw new IOException("Unable to setNull() on prepared statement", e);
+                }
+            } else {
+                try {
+                    ps.setString(index, value.toString());
```

Comment:
Does handling a CLOB using setString() support all required use cases?

Reply:
That's one of the tough things about JDBC and RDBMSs, nobody plays by all the same rules. I tested setString() with PostgreSQL and Oracle and Derby (for the unit tests) and setString() works for CLOB, but setClob() isn't supported for Derby. I'll look at trying setClob() or setCharacterStream() first, and if it fails, fall back to trying setString()

Reply:
Thanks for working through these details, very difficult when JDBC drivers do not behave in a predictable way.  Sounds like a good way forward.